modules/toast/react/lib/Toast.tsx

```diff
@@ -97,13 +97,24 @@ export default class Toast extends React.Component<ToastProps> {
         padding={PopupPadding.s}
         handleClose={onClose}
         closeIconSize={IconButtonSize.Small}
+        role={!onClose && !onActionClick ? 'status' : 'dialog'}
+        aria-live={!onClose && !onActionClick ? 'polite' : 'off'}
+        aria-atomic={!onClose && !onActionClick ? true : false}
         {...elemProps}
       >
         <ToastContentContainer onClose={onClose}>
           {icon && <ToastSystemIcon color={iconColor} colorHover={iconColor} icon={icon} />}
           <Message>
             {this.props.children}
-            {onActionClick && <ActionButton onClick={onActionClick}>{actionText}</ActionButton>}
+            {onActionClick && (
+              <ActionButton
+                data-testid={'action-button'} // used to test in cypress
```

Comment:
Why can't we just use `get/findByRole('button')`? It's one thing to add these to stories, but we should really avoid putting them in our components

Reply:
It's possible that a consumer can have an action link and the close button, both have `button` as the underlying dom element hence why i added a test id to the button. I could rewrite the story so there's only one focusable element in the toast, but then I'm missing out on testing a scenario where there's both a close button and a action button. There's nothing unique about the action button that I could grab it, and I didn't want to grab it by the string `View more details` because that would make the test flaky if we decided to change that text in the story. I agree that adding the test id wasn't the best but I'm open to changing it if I can somehow have both a close button and action link and grab them separately